src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,37 +1,104 @@
 import { useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
-  // Increase the count after successful form submission
-  // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+type Props = {
+  onAdd: (movie: Movie) => void;
+};
+
+const initialForm: Movie = {
+  title: '',
+  description: '',
+  imgUrl: '',
+  imdbUrl: '',
+  imdbId: '',
+};
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
+  const [count, setCount] = useState(0);
+  const [forms, setForms] = useState<Movie>(initialForm);
+
+  const { title, description, imgUrl, imdbUrl, imdbId } = forms;
+
+  const handleFormsChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const { name, value } = event.target;
+
+    setForms(prevForm => ({
+      ...prevForm,
+      [name]: value,
+    }));
+  };
+
+  const reset = () => {
+    setForms({ ...initialForm });
```

Comment:
```suggestion
    setForms(initialForm);
```
You don't have to create a copy